src/diggler/command.d

```diff
@@ -422,16 +424,127 @@ abstract class CommandSet(T) : ICommandSet
 
 	final void registerCommands(T cmdSet)
 	{
-		foreach(memberName; __traits(derivedMembers, T))
+		import diggler.std_backport.meta : staticSort;
+		import std.meta : Filter;
+		import std.algorithm : canFind, commonPrefix, joiner, splitter;
+		import std.range : take, walkLength;
+		import std.string : startsWith, endsWith;
+		import std.conv : to;
+
+		//The separator is strictly a string: any other type will halt compilation
+		static if(__traits(hasMember, T, "subCommandSeparator") && is(typeof(T.subCommandSeparator) == string))
+			enum cmdSep = T.subCommandSeparator;
+		else static if(__traits(hasMember, T, "subCommandSeparator") && !is(typeof(T.subCommandSeparator) == string))
+			static assert(false, "subCommandSeparator must be of type string, not " ~ typeof(T.subCommandSeparator).stringof);
+		else
+			enum cmdSep = "_";
+
+		//Determines whether a given symbol name is a valid command
+		template isCommand(string symbol){
+			static if(__traits(getProtection, __traits(getMember, T, symbol)) != "public")
+				enum isCommand = false;
+			else
+				enum isCommand =
+					symbol != "__ctor" &&
+					symbol != "__dtor" &&
+					__traits(compiles, __traits(getMember, T, symbol)) && // ahem...
+					//__traits(getProtection, __traits(getMember, T, symbol)) == "public" && //needs to be static-if'd
+					isSomeFunction!(mixin("T." ~ symbol)) &&
+					!__traits(isStaticFunction, mixin("T." ~ symbol)) &&
+					!hasAttribute!(mixin("T." ~ symbol), ignore);
+		}
+
+		//Filters a sorted sequence by prefix, e.g. prefixFilter!("a", "a_b", "a_b_c", "d_e_f") yields AliasSeq!("a_b", "a_b_c")
+		template prefixFilter(string prefix, Sequence...)
+		{
+			template prefixFilterImpl(size_t pos, Seq...)
+			{
+				static if(pos >= Seq.length || !Seq[pos].startsWith(prefix))
+					alias prefixFilterImpl = Seq[0..pos];
+				else
+					alias prefixFilterImpl = prefixFilterImpl!(pos+1, Seq);
+			}
+
+			alias prefixFilter = prefixFilterImpl!(0, Sequence);
+		}
+
+		//Basic template to get the first part of a subcommand, e.g. rootCmd("a_b_c") is "a"
+		enum rootCmd(string name) = name.splitter(cmdSep).front.to!string;
```

Comment:
`to!string` should be a no-op here, `name.splitter(cmdSep).front` should be `string` already